### PR TITLE
Feature/issue 170/endpoint conversation history

### PR DESCRIPTION
**Pull request by kirtisynap19**
**Proposed changes**:
- Added new event for reseting the tracker (deleting events as well), and then updating tracker events.
- Added a GET endpoint to get all the messages as part of user conversation
- Updated yaml file
- ...

**Status (please check what you already did)**:
- [ ] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog

### CODE REVIEW

File: rasa_core/events/__init__.py [hunk 1]
```diff
@@ -1038,8 +1044,57 @@ def as_story_string(self):
...
+    Then update tracker events."""
+
+    type_name = "delete_update_events"
+
+    def __init__(self, evts, timestamp=None):
+        self.evts = evts
+        super(DeleteUpdateEvents, self).__init__(timestamp)
+
+    def __hash__(self):
+        return hash(32143124314)
```
Comment (russellwjoel) [inline rasa_core/events/__init__.py]: this hash is always set to this value? Is it used?

File: rasa_core/server.py [hunk 2]
```diff
@@ -822,8 +822,53 @@ def validate_request_body(request: Request, error_message: Text):
...
+        state = tracker.current_state(verbosity)
+        all_events = state["events"]
+
+        # retrieve only 'user' and 'bot' events from the complete list of events
+        user_bot_kv_list = ["user", "bot"]
+        all_user_bot_events = [e for e in all_events if e["event"] in user_bot_kv_list]
+
+        # retrieve only 'events', 'timestamp' and 'text' part of the 'user' and 'bot' events
+        user_bot_keys_list = ["event", "timestamp", "text"]
+        user_bot_events = [dict((key, value) for key, value in a.items() if key in user_bot_keys_list)
```
Comment (russellwjoel) [inline rasa_core/server.py]: 👍  as a thought one could add the switch betweem event and author here.
```
def set_key_value(key):
    if key == 'event':
        return 'author'
    else:
          return key. 
```
Then add this in.   dict (set_key_value(key), value)

**Review by russellwjoel — APPROVED** [review]

File: rasa_core/events/__init__.py (hunk 1 above)
Comment (kirtisynap19) [reply rasa_core/events/__init__.py]: Not really

File: rasa_core/server.py (hunk 2 above)
Comment (kirtisynap19) [reply rasa_core/server.py]: ok. I can do this change :)

**Pull request merged**